### DISCUSSION TITLE
How to make geometrical shapes ?

### DISCUSSION THREAD

**Discussion by RetroAndDev** [CosmosOS/Cosmos#1638] [Q&A]
Hello, i want to make geometrical shapes (like a circle or a rounded rectangle) on cosmos, do you have ideas ?

**Comment by quajak** [comment]
The first step is drawing an arc/circle the code Cosmos uses to draw a circle is here https://github.com/CosmosOS/Cosmos/blob/1e4cbb65e83a800666aba4ca8e244e7585e76594/source/Cosmos.System2/Graphics/Canvas.cs#L331. It uses the Midpoint circle algorith https://en.wikipedia.org/wiki/Midpoint_circle_algorithm.
There is currently no way to draw an arc in cosmos, which should be added. The same algorithm can be used for this. 
A rounded rectangle can be made up of a few arcs and straight lines and I think more complex shapes can be built up from lines and arcs.
